WasatchNET/EEPROM.cs

```diff
@@ -414,122 +422,234 @@ internal EEPROM(Spectrometer spec)
-                maxIntegrationTimeMS    = ParseData.toUInt16(pages[2], 23); //   format >= 5
-                actualPixelsHoriz       = ParseData.toUInt16(pages[2], 25);
-                ROIHorizStart           = ParseData.toUInt16(pages[2], 27);
-                ROIHorizEnd             = ParseData.toUInt16(pages[2], 29);
-                ROIVertRegionStart[0]   = ParseData.toUInt16(pages[2], 31);
-                ROIVertRegionEnd[0]     = ParseData.toUInt16(pages[2], 33);
-                ROIVertRegionStart[1]   = ParseData.toUInt16(pages[2], 35);
-                ROIVertRegionEnd[1]     = ParseData.toUInt16(pages[2], 37);
-                ROIVertRegionStart[2]   = ParseData.toUInt16(pages[2], 39);
-                ROIVertRegionEnd[2]     = ParseData.toUInt16(pages[2], 41);
-                linearityCoeffs[0]      = ParseData.toFloat (pages[2], 43);
-                linearityCoeffs[1]      = ParseData.toFloat (pages[2], 47);
-                linearityCoeffs[2]      = ParseData.toFloat (pages[2], 51);
-                linearityCoeffs[3]      = ParseData.toFloat (pages[2], 55);
-                linearityCoeffs[4]      = ParseData.toFloat (pages[2], 59);
-
-                // deviceLifetimeOperationMinutes = ParseData.toInt32(pages[3], 0);
-                // laserLifetimeOperationMinutes = ParseData.toInt32(pages[3], 4);
-                // laserTemperatureMax  = ParseData.toInt16(pages[3], 8);
-                // laserTemperatureMin  = ParseData.toInt16(pages[3], 10);
-
-                laserPowerCoeffs[0] = ParseData.toFloat(pages[3], 12);
-                laserPowerCoeffs[1] = ParseData.toFloat(pages[3], 16);
-                laserPowerCoeffs[2] = ParseData.toFloat(pages[3], 20);
-                laserPowerCoeffs[3] = ParseData.toFloat(pages[3], 24);
-                maxLaserPowerMW = ParseData.toFloat(pages[3], 28);
-                minLaserPowerMW = ParseData.toFloat(pages[3], 32);
-                laserExcitationWavelengthNMFloat = ParseData.toFloat(pages[3], 36);
-                if (format >= 5)
+                try
                 {
-                    minIntegrationTimeMS = ParseData.toUInt32(pages[3], 40);
-                    maxIntegrationTimeMS = ParseData.toUInt32(pages[3], 44);
+                    baudRate = System.Convert.ToUInt32(a.wrapper.readEEPROMSlot(18));
                 }
+                catch (Exception e)
+                {
+                    baudRate = 0;
+                }
+                hasCooling = false;
+                hasBattery = true;
+                hasLaser = false;
 
-                userData = format < 4 ? new byte[63] : new byte[64];
-                Array.Copy(pages[4], userData, userData.Length);
+                excitationNM = 0;
 
-                badPixelSet = new SortedSet<short>();
-                for (int i = 0; i < 15; i++)
+                //string info = a.wrapper.readEEPROMSlot(15);
+                slitSizeUM = 0;//System.Convert.ToUInt16(info.Substring(5));

```

Comment:
Instead of substring, it might have been a packed byte field.  Don't remember offhand.